src/beaconer.rs

```diff
@@ -213,21 +209,21 @@ impl Beaconer {
         if self.disabled {
             return;
         }
-        match self.mk_beacon().await {
-            Ok(beacon) => {
-                self.send_beacon(beacon).await;
-                // On success just use the normal behavior for selecting a next
-                // beacon time. Can't be the first time since we have region
-                // parameters to construct a beacon
-                self.next_beacon_time = Self::mk_next_beacon_time(self.interval);
-            }
-            Err(err) => {
-                warn!(%err, "construct beacon");
-                // On failure to construct a beacon at all, select a shortened
-                // "first time" next beacon time
-                self.next_beacon_time = Self::mk_next_short_beacon_time(self.interval);
-            }
-        };
+        let interval = self.interval;
+        let (last_beacon, next_beacon_time) = self
+            .mk_beacon()
+            .inspect_err(|err| warn!(%err, "construct beacon"))
+            .and_then(|beacon| self.send_beacon(beacon))
+            // On success to construct and transmit a beacon and it's report
+            // select a normal full next beacon time
+            .map_ok(|beacon| (Some(beacon), Self::mk_next_beacon_time(interval)))
+            // On failure to construct, transmit or send a beacon or it's
```

Comment:
While we're here .... `s/it's/its/`

Reply:
fixed in [c13647b](https://github.com/helium/gateway-rs/pull/434/commits/c13647b1071e8cf1475b52f66b792846e89f2814)